### DISCUSSION TITLE
Marketing stream- Step 3: Subscribe to Amazon Marketing Stream datasets

### DISCUSSION THREAD

**Discussion by giangdaughtry** [amzn/ads-advanced-tools-docs#153] [General]
Hi, I am facing trouble in Subscribing to sponsored ads datasets. After I send POST request, I receive {"message":"Unauthorized"}
Could anybody help me?

Thank you.

**Comment by elizafarley** [comment]
HI @giangdaughtry, have you already completed the [steps to onboard](https://advertising.amazon.com/API/docs/en-us/guides/onboarding/overview) to the Amazon Ads API? If you don't have a valid profile/auth token, you won't be able to make a successful POST request.

**Reply by giangdaughtry** [reply]
Hello @elizafarley, thank you so much for your response.
I already did step 3 Subscribe to Amazon Marketing Stream datasets successfully.
Im in progress with step 4.
